### ISSUE TITLE
Language server seems to crash when creating a new `.malloy` file

### ISSUE THREAD

**Issue by christopherswenson** [looker-open-source/malloy#359]
Steps to reproduce:

1. Run VSCode Extension
2. Open an existing `.malloy` file and wait for it to syntax highlight
3. Command+N 
4. Type/paste some Malloy code
5. Save as `something.malloy`
6. Notice that it does not highlight

**Comment by christopherswenson** [comment]
According to `git bisect`
```
79f0b6df5bdb1e39a00afb3a3d16f42ae68096ee is the first bad commit
commit 79f0b6df5bdb1e39a00afb3a3d16f42ae68096ee
Author: Michael Toy <mtoy@google.com>
Date:   Sat Jan 29 11:05:39 2022 -1000

    Add parser supper for sql: command

 packages/malloy/src/lang/ast/ast-main.ts    |  31 +-
 packages/malloy/src/lang/grammar/Malloy.g4  |  16 +
 packages/malloy/src/lang/parse-malloy.ts    | 604 ++++++++++++++++------------
 packages/malloy/src/lang/parse-to-ast.ts    |  14 +-
 packages/malloy/src/lang/parse.spec.ts      |  20 +
 packages/malloy/src/lang/test-translator.ts |   2 +-
 packages/malloy/src/lang/zone.ts            |  36 +-
 7 files changed, 461 insertions(+), 262 deletions(-)
```

That is https://github.com/looker-open-source/malloy/commit/79f0b6df5bdb1e39a00afb3a3d16f42ae68096ee by @mtoy-googly-moogly

**Comment by christopherswenson** [comment]
By stepping through the language server code, I found that it's crashing on the line
```
await runtime.getModel(new URL(uri));
```
in `getMalloyDiagnostics`

**Comment by christopherswenson** [comment]
Ah. That's because it's not a crash at all. It's an infinite loop.

**Comment by mtoy-googly-moogly** [comment]
Fixed in #360 and test added in #361
